src/main/java/io/xcherry/core/client/BasicClient.java

```diff
@@ -124,6 +131,52 @@ public <T> T invokeRPC(final ProcessExecutionRpcRequest request, final Class<T>
         return clientOptions.getObjectEncoder().decodeFromEncodedObject(response.getOutput(), returnType);
     }
 
+    /**
+     * Wait for a process execution to complete within the specified timeout seconds.
+     *  1. If the process execution is still running after the timeout seconds, {@link WaitTimeoutException} will be thrown.
+     *  2. If the process execution has stopped by system (e.g., due to re-balancing in the server), {@link ServerRuntimeException} will be thrown.
+     *  3. In other cases, the current status of the process execution will be returned.
+     *
+     * @param namespace         the namespace in which the operation should be performed.
+     * @param processId         a unique identifier used to differentiate between different executions of the same process type.
+     * @param timeoutInSeconds  a value less than or equal to 30.
```

Comment:
I thought we want to have an overloaded method to not set the timeout

Reply:
We do have such method in Client. [link](https://github.com/xcherryio/sdk-java/pull/18/files#diff-81d639b14bb19030c4d42d99bf4bf8e2cfed82e178f8832919c048d69e457f18R352).

Here in BasicClient, we usually do not set default value for methods. Users who use BasicClient will configurate the needed parameters. E.g., the [stopProcess](https://github.com/xcherryio/sdk-java/blob/main/src/main/java/io/xcherry/core/client/BasicClient.java#L60) method.